src/azul/service/avro_pfb.py

```diff
@@ -91,18 +92,18 @@ def add_doc(self, doc: JSON):
         file_relations = set()
         for entity_type, entities in contents.items():
             if entity_type != 'files':
-                for entity in entities:
-                    if 'document_id' in entity:
-                        entity = PFBEntity.from_json(name=entity_type,
-                                                     object_=entity,
-                                                     schema=self.schema)
-                        if entity not in self._entities:
-                            self._entities[entity] = set()
-                        file_relations.add(PFBRelation.to_entity(entity))
-                    else:
-                        # FIXME: Protocol entities lack document ID so we skip for now
-                        #        https://github.com/DataBiosphere/azul/issues/3084
-                        pass
+                # FIXME: Protocol entities lack document ID so we skip for now
+                #        https://github.com/DataBiosphere/azul/issues/3084
+                entities = (e for e in entities if 'document_id' in e)
+                # Sorting entities is required for deterministic output since
+                # contributions could of been aggregated in any random order.
```

Comment:
```suggestion
                # Sorting entities is required for deterministic output since
                # the order of the inner entities in an aggregate document is tied to the order with which contributions are returned by ES during aggregation, which happens to be non-deterministic.
```